cdc/generation_id.hh

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Modified by ScyllaDB
+ * Copyright (C) 2021 ScyllaDB
+ *
+ */
+
+#pragma once
+
+#include <variant>
+
+#include "db_clock.hh"
+
+namespace cdc {
+
+struct generation_id {
+    db_clock::time_point ts;
+};
+
+std::ostream& operator<<(std::ostream&, const generation_id&);
+bool operator==(const generation_id&, const generation_id&);
+db_clock::time_point get_ts(const generation_id&);
```

Comment:
Why not make this a member function?

Reply:
Because it can only be a member function if we assume that `generation_id` is defined by us (and cannot e.g. when it's a std type). In #8286 I change the definition of `generation_id` to be a `std::variant`, and thanks to this being a free function earlier, I don't need to change the places where I use `get_ts` (I only need to change the definition of `get_ts`).